modules/mem_application/src/mem_application.cpp

```diff
@@ -11,48 +11,40 @@
 int main(int argc, char **argv) {
 
   if (argc != 3) {
-    spdlog::error("MemApp: Introduce number of images and configuration path");
+    spdlog::error("MemApp: Pass number of iterations and rb_config.yaml path");
     return 0;
   }
   int iters = std::stoi(argv[1]);
-  std::string yaml_path = argv[2];
+  std::string yamlFile = argv[2];
 
-  // 1.- Load config
+  kpsr::YamlEnvironment yamlEnv(yamlFile);
 
-  kpsr::YamlEnvironment yamlEnv(yaml_path + "/image.yaml");
-
-  float imgWidth;
-  float imgHeight;
+  int imgWidth;
+  int imgHeight;
   int roboBeePrefix = 1;
 
-  yamlEnv.getPropertyFloat("img_width", imgWidth);
-  yamlEnv.getPropertyFloat("img_height", imgHeight);
-
-  spdlog::info("MemApp: loaded width: {:03.0f}", imgWidth);
-  spdlog::info("MemApp: loaded height: {:03.0f}", imgHeight);
+  yamlEnv.getPropertyInt("img_width", imgWidth);
+  yamlEnv.getPropertyInt("img_height", imgHeight);
 
-  // 2.- Create EventLoop
+  spdlog::info("MemApp: loaded width: {}", imgWidth);
+  spdlog::info("MemApp: loaded height: {}", imgHeight);
 
   kpsr::high_performance::EventLoopMiddlewareProvider<16> eventloop(nullptr);
 
   eventloop.start();
 
-  // 3.- Publishers and subscribers
 
   kpsr::Subscriber<kpsr::vision_ocv::ImageData> *imageDataSubscriber =
       eventloop.getSubscriber<kpsr::vision_ocv::ImageData>("ImageData");
   kpsr::Publisher<kpsr::vision_ocv::ImageData> *imageDataPublisher =
       eventloop.getPublisher<kpsr::vision_ocv::ImageData>(
-          "ImageData", 0, nullptr,
-          nullptr); // 32 pool, (imgFactory) initFunc, clonerFunc
+          "ImageData", 32, nullptr, nullptr);
```

Comment:
Why remove the tuning functions? Please add them back...

Reply:
Added both functions but following error is appearing in runOnce() functions:
[2019-12-24 03:33:44.840] [info] MemApp: RoboBee Service started
[2019-12-24 03:33:44.840] [info] MemApp: QueenBee Service started
mem_application: /home/javi/thirdparties/klepsydra2/kpsr-robotics/vision_ocv/modules/vision_ocv/src/image_data_factory.cpp:26: void kpsr::vision_ocv::ImageDataFactory::eventCloner(const kpsr::vision_ocv::ImageData&, kpsr::vision_ocv::ImageData&): Assertion `target.img.type() == source.img.type()' failed.
Aborted (core dumped)